src/components/FetchComponent.js

```diff
@@ -0,0 +1,71 @@
+import React, { Component } from "react";
+
+class FetchComponent extends Component {
+  constructor(props) {
+    super(props);
+    this.fetchJSON = this.fetchJSON.bind(this);
+    /* Placeholder table until we load the data */
+    this.state = {
+      data: {
+        identity: {
+          id: null,
+          login: null,
+        },
+        permissions: {
+          roles: null,
+        },
+      },
+    };
+  }
+  /* Fetches json from mocky API */
+  async fetchJSON() {
+    try {
+      const response = await fetch(
+        "https://run.mocky.io/v3/02e9fb29-4516-48b1-8bd2-43e868303cb6"
+      );
+      const json = await response.json();
+      this.setState({ data: json });
+    } catch (error) {
+      console.log(error);
+    }
+    /* For debugging */
+    console.log(this.state.data);
+  }
+
+  render() {
+    return (
+      <div>
+        <h1>{this.props.title}</h1>
+        <button onClick={this.fetchJSON}>Get data</button>
+        <table border={2} cellPadding={5}>
```

Comment:
Not showing the user that data is being loaded.
As data will come asynchronously you need to show the user that you are waiting for data.

Reply:
I've added a loading indicator while the function is fetching the JSON data in the renamed [`UserTable`](https://github.com/jql6/mentorsNetwork-FrontEnd/blob/FetchAPI/src/components/UserTable.js) component file.

This is done through a `loading` status which is set to `true` in `loadData()` before the json file is obtained, and set back to `false` once the file is obtained.
![Loading](https://user-images.githubusercontent.com/46179386/122305680-22537380-cebc-11eb-9162-7a1d795a5feb.png)

When the `loading` status is `true`, the element displays "Loading...".
![loading 2](https://user-images.githubusercontent.com/46179386/122305571-fc2dd380-cebb-11eb-853a-de9c9f068868.png)